ouroboros-network/src/Ouroboros/Network/NodeToClient.hs

```diff
@@ -231,3 +246,108 @@ withServer_V1 muxTracer handshakeTracer tbl addr peeridFn versionData applicatio
         versionData
         (DictVersion nodeToClientCodecCBORTerm)
         application)
+
+-- | 'ncSubscriptionWorker' which starts given application versions on each
+-- established connection.
+--
+ncSubscriptionWorker
+    :: forall appType peerid void x y.
+       ( HasInitiator appType ~ True
+       , Show peerid )
+    => Tracer IO (WithIPList (SubscriptionTrace Socket.SockAddr))
+    -> Tracer IO (WithMuxBearer (MuxTrace NodeToClientProtocols))
+    -> Tracer IO
+        (TraceSendRecv
+          (Handshake NodeToClientVersion CBOR.Term)
+          peerid
+          (DecoderFailureOrTooMuchInput DeserialiseFailure))
+    -> (Socket.SockAddr -> Socket.SockAddr -> peerid)
+    -> ConnectionTable IO Socket.SockAddr
+    -> LocalAddresses Socket.SockAddr
```

Comment:
Does this want to be the full `LocalAddresses` since we are only going to be connecting to a local socket, so there's no local address in this case right?

Reply:
I don't see why we have to limit it to local sockets, the rest of NodeToClient works over any socket.